cassandra/cluster.py

```diff
@@ -1762,6 +1764,8 @@ def connect(self, keyspace=None, wait_for_all_pools=False):
                 try:
                     self.control_connection.connect()
 
+                    self.load_balancing_policy.populate(weakref.proxy(self), self.metadata.all_hosts())
```

Comment:
Not sure if this code was here before most recent pushes, I don't remember seeing it.
Why do you need to pass weakref? Is the cluster saved somewhere?
Why do we need to call this here? I see that populate is called in other places already, so why another call here?

Reply:
When we were using experimental setting, it was known from the beginning that we are using tablets. So, in every place where `self.load_balancing_policy.populate()` was called, it was properly informed to use tablets. But now, we have to get the information about using tablets from connection handshake. Therefore, when we are performing populate on lbp here: https://github.com/scylladb/python-driver/blob/master/cassandra/cluster.py#L1744, we do not know yet if tablets are being used or not, so there is a need to call `populate` again after calling `connect()` on `control_connection`.

Reply:
Ok, but please put proper explanation in the comment, this code is really not obvious

Reply:
@sylwiaszunejko It looks like this line specifically causes the CI failures. Since pretty much all integration tests don't use tablets (and your code shouldn't affect them), I tested progressively removing more and more of your changes and removing this line fixed the failing tests. I don't yet have an explanation why it is.

Reply:
Branch on which I tested: https://github.com/avelanarius/python-driver/commits/tablets_disabled3/, "no populate" fixed the failing tests ("Build and upload to PyPi" are failing for some other reason)